drs_cli/client.py

```diff
@@ -1,195 +1,424 @@
+        try:
+            response = requests.delete(
+                url=url,
+                headers=self.headers,
+            )
+        except (
+            requests.exceptions.ConnectionError,
+            socket.gaierror,
+            urllib3.exceptions.NewConnectionError,
+        ):
+            raise requests.exceptions.ConnectionError(
+                "Could not connect to API endpoint."
+            )
+        if not response.status_code == 200:
+            try:
+                response_val = Error(**response.json())
+            except (
+                json.decoder.JSONDecodeError,
+                pydantic.ValidationError,
+            ):
+                raise InvalidResponseError(
+                    "Response could not be validated against API schema."
+                )
+            logger.warning("Received error response.")
         else:
-            return Error(**response.json())
+            try:
+                response_val = str(response.json())
+            except TypeError:
+                raise InvalidResponseError(
+                    "Response could not be validated against API schema."
+                )
+            logger.info(f"Object deleted: {object_id}")
+        return response_val
 
-    def _get_headers(self):
-        """Set dictionary of request headers in `self.headers`."""
-        self.headers: Dict = {
+    def _get_headers(self) -> Dict:
+        """Build dictionary of request headers."""
+        headers: Dict = {
             'Content-type': 'application/json',
         }
         if self.token:
-            self.headers['Authorization'] = 'Bearer ' + self.token
+            headers['Authorization'] = 'Bearer ' + self.token
+        return headers
+
+    def _get_host(
+        self,
+        uri: str,
+    ) -> Tuple[str, str]:
+        """Extract URI schema and domain or IP from HTTP, HTTPS or DRS URI.
+
+        Arguments:
+            uri: HTTP or HTTPS URI pointing to the root domain/IP of a DRS
+                instance OR a hostname-based DRS URI to a given object, cf.
+                https://ga4gh.github.io/data-repository-service-schemas/preview/develop/docs/#_hostname_based_drs_uris.
+                Anything after a slash following the domain/IP will be ignored.
+
+        Returns:
+            Tuple of URI schema (e.g., 'https', 'drs') and host domain or IP
+            (e.g., 'my-drs.app', '0.0.0.0').
+
+        Raises:
+            drs_cli.errors.InvalidURI: input URI cannot be parsed.
+
+        Examples:
+           >>> DRSClient.get_host(uri="https://my-drs.app/will-be-ignored")
+           ('https', 'my-drs.app')
+           >>> DRSClient.get_host(uri="drs://my-drs.app/My0bj3ct")
+           ('drs', 'my-drs.app')
+        """
+        match = re.search(self._RE_HOST, uri, re.I)
+        if match:
+            schema = match.group('schema')
+            host = match.group('host').rstrip('\\')
+            if len(host) > 253:
+                raise InvalidURI
```

Comment:
This condition won't happen ever, right?

Because above the regex is `_RE_DOMAIN_PART = r'[a-z0-9]([a-z0-9-]{1,61}[a-z0-9])?'` which allows only 61 charater repetition?

Reply:
Note that the capture group `host` is defined by `_RE_DOMAIN = rf"({_RE_DOMAIN_PART}\.)+{_RE_DOMAIN_PART}\.?"` which contains multiple repetitions of `_RE_DOMAIN_PART`. So yeah, each part (like `google` and `com` in `google.com`) can be at most 63 characters (first and last character cannot be hyphens), but all together cannot exceed 253 chars.